Pocket/pocket-ios
# Premium Reader Settings

## Summary
This PR adds new Reader settings for premium users
* Premium fonts
* Line height
* Margins
* Text Alignment (originally not part of the PRD)
* Reset to defaults
## References 
* Premium settings PRD

## Implementation Details
* Update `FontDescriptor`, add premium font families
* Add `GenericfontStyling`, concrete implementation of `FontStyling` for a generic font family
* Update `ReaderSettings`, `ReaderSettingsView` to reflect the above changes

## Test Steps
* Obtain the complete font set and install them in `PocketKit/Sources/Textile/Style/Typography/Fonts`
* Build/run this branch
* Login with a free account
* Go to reader/display settings and make sure you only see two fonts: Blanco and Graphik, and you can scale their size
* Make sure you see the premium upsell button and tapping on it opens the premium upgrade screen
* Logout and re-login with a premium account, make sure you see the additional premium features listed above
* Make sure all the features work as expected

## PR Checklist:

## Animations
iPhone - Premium | iPad - Free
-- | --
<img width=200 src=https://github.com/Pocket/pocket-ios/assets/34376330/7550b10f-9168-4e01-acc9-2bc3176f995f> | <img width=500 src=https://github.com/Pocket/pocket-ios/assets/34376330/e4d92e68-d4ca-4304-b36b-f408ac9c0dfe>
